### PR TITLE
DOC: some reviewing of the 0.20 whatsnew file

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Taking the opportunity to do some further edits to the whatsnew file

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Mainly some rewording to make things a bit clearer (IMO), and some sphinx fixes.

I also added the warning about IntervalIndex being experimental that were in the main docs also in the whatsnew file

File: doc/source/whatsnew/v0.20.0.txt [hunk 1]
```diff
@@ -221,33 +210,24 @@ Using an explicit compression type
...
-Inferring compression type from the extension
-
-.. ipython:: python
-
-   df.to_pickle("data.pkl.xz", compression="infer")
-   rt = pd.read_pickle("data.pkl.xz", compression="infer")
-   rt
-
-The default is to ``infer``:
+The default is to inferring compression type from the extension (``compression='infer'``):
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.20.0.txt]: "is to inferring" -> "is to infer"

File: doc/source/whatsnew/v0.20.0.txt [hunk 2]
```diff
@@ -411,6 +391,11 @@ pandas has gained an ``IntervalIndex`` with its own dtype, ``interval`` as well
 notation, specifically as a return type for the categories in :func:`cut` and :func:`qcut`. The ``IntervalIndex`` allows some unique indexing, see the
 :ref:`docs <indexing.intervallindex>`. (:issue:`7640`, :issue:`8625`)
 
+.. warning::
+
+   These indexing behaviors of the IntervalIndex are provisional and may change in a future version of pandas. Feedback on usage very welcome.
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.20.0.txt]: "*is* very welcome" (or just "is welcome")

**Review by TomAugspurger — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@TomAugspurger added a commit with a few further updates. Can you take a quick look? (travis seems ok)

**Review by TomAugspurger — APPROVED** [review]

**Pull request merged**